test/unit_test/engine/packaging/test_packaging_generator.py

```diff
@@ -215,6 +224,116 @@ def test_generate_zipfile_artifacts_zero_len_nodes(tmp_path):
+    from azure.ai.ml.constants import AssetTypes
+    from azure.ai.ml.entities import Model
+    from azure.core.exceptions import ServiceResponseError
+
+    version = "1.0"
+    description = "Test description"
+    name = "OutputModels"
+    model_id = "model_id"
+
+    packaging_config = PackagingConfig(
+        type=PackagingType.AzureMLModels,
+        config=AzureMLModelsPackagingConfig(version=version, description=description),
+        name=name,
+    )
+
+    model_path = "fake_model_file"
+
+    footprints = get_footprints(model_id, model_path)
+
+    azureml_client_config = Mock(max_operation_retries=3, operation_retry_interval=5)
+    ml_client_mock = Mock()
+    azureml_client_config.create_client.return_value = ml_client_mock
+    resource_path_mock = Mock()
+    mock_create_resource_path.return_value = resource_path_mock
+
+    model = Model(
+        path=model_path,
+        type=AssetTypes.CUSTOM_MODEL,
+        name=name,
+        version=version,
+        description=description,
+    )
+
+    # execute
+    generate_output_artifacts(
+        packaging_config, footprints, footprints, output_dir=None, azureml_client_config=azureml_client_config
+    )
+
+    # assert
+    assert mock_retry_func.call_once_with(
+        ml_client_mock.models.create_client,
+        [model],
+        max_tries=azureml_client_config.max_operation_retries,
+        delay=azureml_client_config.operation_retry_interval,
+        exceptions=ServiceResponseError,
+    )
+
+
+@patch("olive.engine.packaging.packaging_generator.retry_func")
+@patch("olive.engine.packaging.packaging_generator.create_resource_path")
+def test_generate_azureml_data(mock_create_resource_path, mock_retry_func):
+    from azure.ai.ml.constants import AssetTypes
+    from azure.ai.ml.entities import Data
+    from azure.core.exceptions import ServiceResponseError
+
+    version = "1.0"
+    description = "Test description"
+    name = "OutputModels"
+    model_id = "model_id"
+
+    packaging_config = PackagingConfig(
+        type=PackagingType.AzureMLData,
+        config=AzureMLDataPackagingConfig(version=version, description=description),
+        name=name,
+    )
+
+    model_path = "fake_model_file"
+
+    footprints = get_footprints(model_id, model_path)
+
+    azureml_client_config = Mock(max_operation_retries=3, operation_retry_interval=5)
+    ml_client_mock = Mock()
+    azureml_client_config.create_client.return_value = ml_client_mock
+    resource_path_mock = Mock()
+    mock_create_resource_path.return_value = resource_path_mock
+
+    data = Data(
+        path=model_path,
+        type=AssetTypes.URI_FILE,
+        name=name,
+        version=version,
+        description=description,
+    )
+
+    # execute
+    generate_output_artifacts(
+        packaging_config, footprints, footprints, output_dir=None, azureml_client_config=azureml_client_config
+    )
+
+    # assert
+    assert mock_retry_func.call_once_with(
```

Comment:
do we need add integration test to check the uploaded contents is like what we want?
Originally, the rank feature seems incomplete since there is no integration test to check the zip contents.

Reply:
Yeah sure, I will add this later. I've created a workitem to track this: https://aiinfra.visualstudio.com/Model%20optimization%20Toolkit/_workitems/edit/27261